### PR TITLE
Use cross stripper when cross compiling and allow overriding native s…

### LINKED ISSUE #1414: cross compilation doesn't support the "--strip" option
cross_file.txt:
c = '/opt/br_mips-2013.11/usr/bin/mips-linux-gcc'
cpp = '/opt/br_mips-2013.11/usr/bin/mips-linux-g++'
ar = '/opt/br_mips-2013.11/usr/bin/mips-linux-ar'
strip = '/opt/br_mips-2013.11/usr/bin/mips-linux-strip'

[host_machine]
system = 'linux'
cpu_family = 'mips'
cpu = 'mips'
endian = 'big'

Without the "-strip" option it works fine:
meson --cross-file ../../../../lte1160/cross_file.txt . $d --prefix=$HOME/tmp/meson
ninja-build -C build
ninja-build -C build install

With the "-strip" option it doesn't work:
meson --cross-file ../../../../lte1160/cross_file.txt . $d --prefix=$HOME/tmp/meson --strip
ninja-build -C build
ninja-build -C build install

Stripping target 'libkt_dmc_cm-1.0.so'
Could not strip file.

Stdout:

Stderr:
strip: Unable to recognise the format of the input file `/home/fabiopo/tmp/meson/lib64/libkt_dmc_cm-1.0.so'

FAILED: install 
'/usr/bin/python3' '/home/fabiopo/packages/meson/meson.py' '--internal' 'install' '/ssd/lte/app-2/tcommon/src/oma-dm/kt_lib/build/meson-private/install.dat'
ninja: build stopped: subcommand failed.

I'm doing something wrong? It is expected? Any suggestion on how to fix it?

Thanks

### PR DESCRIPTION
**Pull request by jpakkane**
…trip executable. Closes #1414.

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -592,9 +592,19 @@ def generate_coverage_rules(self, outfile):
 
     def generate_install(self, outfile):
         install_data_file = os.path.join(self.environment.get_scratch_dir(), 'install.dat')
+        if self.environment.is_cross_build():
+            bins = self.environment.cross_info.config['binaries']
+            if 'strip' not in bins:
+                mlog.warning('Cross file does not specify strip binary, result will not be stripped.')
+                strip_bin = None
+            else:
+                strip_bin = mesonlib.stringintlistify(bins['strip'])
```
Comment (nirbheek) [inline mesonbuild/backend/ninjabackend.py]: `stringintlistify` will accept both `int` and `str`, we don't want to use that here. We want to use `stringlistify` here.

File: mesonbuild/backend/ninjabackend.py [hunk 2]
```diff
@@ -592,9 +592,19 @@ def generate_coverage_rules(self, outfile):
 
     def generate_install(self, outfile):
         install_data_file = os.path.join(self.environment.get_scratch_dir(), 'install.dat')
+        if self.environment.is_cross_build():
```
Comment (nirbheek) [inline mesonbuild/backend/ninjabackend.py]: We also support installing `native: true` build targets, don't we? So maybe we should check for that too?

Or just disallow `install: true` on targets that are `native: true` in `BuildTarget().process_kwargs()` which seems more sane to me.

**Comment by fabio-porcedda** [comment]
Using your branch the "--strip" option works fine even for cross compile!

Thanks

File: mesonbuild/backend/ninjabackend.py (hunk 2 above)
Comment (jpakkane) [inline mesonbuild/backend/ninjabackend.py]: Disallowed.

**Pull request merged**